tests/test_cmdline.py

```diff
@@ -31,6 +33,45 @@ def test_sig(sig_name):
         [helpers.path_to_executable('test_sig'), sig_name],
     )
 
+# Stateful signature schemes testing
+@helpers.filtered_test
+@pytest.mark.parametrize('sig_stfl_name', helpers.available_sig_stfls_by_name())
+def test_sig_stfl(sig_stfl_name):
+    if not(helpers.is_sig_stfl_enabled_by_name(sig_stfl_name)): pytest.skip('Not enabled')
+    helpers.run_subprocess(
+        [helpers.path_to_executable('test_sig_stfl'), sig_stfl_name],
+    )
+
+# Stateful signature schemes testing with key read
+@helpers.filtered_test
+@pytest.mark.parametrize('sig_stfl_name', helpers.available_sig_stfls_by_name())
+def test_sig_stfl_read_key(sig_stfl_name):
+    directory = "../tests/saved_keys/"
+    filename = "tmp_" + sig_stfl_name
+    if not(helpers.is_sig_stfl_enabled_by_name(sig_stfl_name)): pytest.skip('Not enabled')
+    helpers.run_subprocess(
+        [helpers.path_to_executable('test_sig_stfl'), sig_stfl_name, "--read_key_from="+directory+filename],
+    )
+
+# Stateful signature schemes testing with key save
+@helpers.filtered_test
+@pytest.mark.parametrize('sig_stfl_name', helpers.available_sig_stfls_by_name())
+def test_sig_stfl_save_key(sig_stfl_name):
+    directory = "../tests/saved_keys/"
+    
+    # Extract parts of the scheme name to form the filename
+    height_of_tree = sig_stfl_name[sig_stfl_name.find('_')+1:sig_stfl_name.rfind('_')].replace('/', '')
+    hash_alg = sig_stfl_name[sig_stfl_name.find('-')+1:sig_stfl_name.find('_')].lower()
+    sig_scheme = sig_stfl_name[0:sig_stfl_name.find('-')].lower()
+    hash_alg_bits = sig_stfl_name[sig_stfl_name.rfind('_')+1:]
```

Comment:
This is dependent on the current naming conventions of XMSS but wouldn't necessarily apply to LMS. Is there a reason you need to extract all these fields? Can't you just use the scheme name directly to construct the filename, without parsing?

Reply:
Yes, I should have done that when I started out saving the files. I wil mark this for removal.

Reply:
Once all the circleCI tests pass, I will rename the saved keys and then remove this parsing code